proxy/src/main.rs

```diff
@@ -118,11 +119,32 @@ async fn main() -> anyhow::Result<()> {
     let mgmt_address: SocketAddr = arg_matches.value_of("mgmt").unwrap().parse()?;
     let http_address: SocketAddr = arg_matches.value_of("http").unwrap().parse()?;
 
+    // determine common name from tls-cert (-c server.crt param).
+    // used in asserting project name formatting invariant.
+    let common_name = match arg_matches.value_of("tls-cert") {
+        Some(cert_path) => {
+            let file = std::fs::File::open(cert_path)?;
+            let almost_common_name = x509_parser::pem::Pem::read(std::io::BufReader::new(file))?
+                .0
+                .parse_x509()?
+                .tbs_certificate
+                .subject
+                .to_string();
+            let prefix = "CN=*";
+            // making sure formatting is as expected: almost_common_name = "'CN=*'[<common name>]"
+            assert_eq!(almost_common_name.substring(0, prefix.len()), prefix);
+            let common_name = almost_common_name.substring(prefix.len(), almost_common_name.len());
```

Comment:
This doesn't look natural to me. See https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix

Reply:
will refactor!